packages/flutter/lib/src/widgets/container.dart

```diff
@@ -434,6 +443,17 @@ class Container extends StatelessWidget {
     if (transform != null)
       current = Transform(transform: transform, child: current);
 
+    if (clipBehaviour != null &&
+        clipBehaviour != Clip.none &&
+        decoration is BoxDecoration) {
```

Comment:
It's strange that the clipping will only work for BoxDecorations. What about a ShapeDecoration? Or any other decoration?

Reply:
we need to access the borderRadius inside the decoration
- We can just check for BoxDecoration ( Wich will support must use cases )
- Or We can create interface like ShapeWithBorder and add it to the all decorations
- Or we can just add another property in the Container as 'borderRadius' and use that

Wich one do you think is the best? @goderbauer

Reply:
We should ensure that this works for all decorations. We probably need to add something to the Decoration base class to get a clipPath (similar to ShapeBorder.getOutterPath) and then use that here.